js/thread-client.js

```diff
@@ -0,0 +1,34 @@
+const { DebuggerClient } = require('devtools/shared/client/main');
+const { DebuggerTransport } = require('devtools/transport/transport');
+const { TargetFactory } = require("devtools/client/framework/target");
+const promise = require('devtools/sham/promise');
+
+const socket = new WebSocket("ws://localhost:9000");
+const transport = new DebuggerTransport(socket);
+const client = new DebuggerClient(transport);
+
+function connectToClient(onConnect) {
+  client.connect().then(() => {
+    return client.listTabs().then(onConnect);
+  }).catch(err => console.log(err));
+}
+
+function connectToTab(tab, onNewSource) {
+  let deferred =  promise.defer();
+  const options = { client, form: tab, chrome: false };
+
+  TargetFactory.forRemoteTab(options).then(target => {
+    target.activeTab.attachThread({}, (res, threadClient) => {
+      threadClient.resume();
+      window.gThreadClient = threadClient;
```

Comment:
Yeah, sounds good. Let's talk next week about this.

Can you rename the file to "client.js" or something like that though? Because it creates the entire connection, not just the thread client. Eventually we will need the `DebuggerClient` for some things too, I think. You could name this `connection.js`, I dunno.